### ISSUE TITLE
[BUG] [Linter] Linter doesn't raise H025 when there are matching pairs of orphan closing tags

### ISSUE THREAD

**Issue by MrkGrgsn** [djlint/djLint#786]
- [X] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [X] I've searched the [issues](https://github.com/Riverside-Healthcare/djLint/issues)
- [X] I've read the [docs](https://djlint.com)

## System Info
 - OS: Fedora 38
 - Python Version 3.11.6
 - djLint Version 1.34.0
 - template language: N/A

## Issue
If there are matching orphaned closing HTML tags, an H025 warning isn't raised. I expect it to raise a warning for every orphan closing tag.

The rule inserts orphaned closing tags into the list of opening tags, paving the way for subsequent closing tags to match earlier closing tags and treating them as closed https://github.com/Riverside-Healthcare/djLint/blob/master/src/djlint/rules/H025.py#L49

## How To Reproduce
A minimal example is this HTML source:
```html
</p>
</p>
```
If you remove one of the tags, H025 is raised as expected. It happens in more complex and realistic scenarios too:
```html
</p>
<h1>blah blah blah</h1>
<p>blah blah blah</p>
</p>
```
and
```html
<h1>blah blah blah</p>
<p>blah blah blah</p>
</p>
```

## Contents of .djlintrc/pyproject.toml [tool.djlint]
```yaml
{
    "profile": "django",
    "indent": 2,
    "ignore": "T003,H021"
}
```
